### PR TITLE
Fix all ESLint issues

### PR DESCRIPTION
**Pull request by lexisother**
Does as it says on the tin.

### CODE REVIEW

**Comment by lleyton** [comment]
> merge 107 commits into [main](https://github.com/replugged-org/replugged)

uuuh, I don't think that's that you want, lexi-cat

**Comment by lexisother** [comment]
silence, wench!

**Review by lleyton — APPROVED** [review]
Ready.

File: scripts/inject/index.ts [hunk 1]
```diff
@@ -71,7 +71,7 @@ let platform: DiscordPlatform | undefined;
             platform = current;
             break;
           }
-        } catch (e) {}
+        } catch (_e) {}
```
Comment (asportnoy) [inline scripts/inject/index.ts:74]: Let's just remove the catch argument for all of these cases (not gonna comment on each one).
```suggestion
        } catch {}
```

File: src/globals.d.ts [hunk 2]
```diff
@@ -10,9 +10,10 @@ declare global {
   };
 
   interface Window {
-    React: any;
+    React: typeof import("react");
```
Comment (asportnoy) [inline src/globals.d.ts:13]: Add import `import type React from "react";` above and change this to `React: typeof React;`

File: src/preload.ts [hunk 3]
```diff
@@ -78,7 +78,8 @@ export type RepluggedNativeType = typeof RepluggedNative;
 contextBridge.exposeInMainWorld("RepluggedNative", RepluggedNative);
 
 const renderer = ipcRenderer.sendSync(RepluggedIpcChannels.GET_RENDERER_JS);
-webFrame.executeJavaScript(renderer);
+// eslint-disable-next-line @typescript-eslint/no-floating-promises
+webFrame.executeJavaScript(renderer).then((_r) => {});
```
Comment (asportnoy) [inline src/preload.ts:82]: Why do you need the promise here? Might be good to add a comment explaining.

File: src/types/addon.ts [hunk 4]
```diff
@@ -10,6 +10,7 @@ export type Id = z.infer<typeof id>;
 
 export const author = z.object({
   name: z.string(),
+  // eslint-disable-next-line @typescript-eslint/naming-convention
   discord_id: z.string(),
```
Comment (asportnoy) [inline src/types/addon.ts:14]: Should we switch this parameter to camel case instead?

**Review by asportnoy — CHANGES_REQUESTED** [review]
Pretty good for the most part, just some minor things here and there.

**Review by asportnoy — APPROVED** [review]
LGTM!

**Comment by lleyton** [comment]
:leaSpin:

**Pull request merged**